internal/helm/envresolver.go

```diff
@@ -64,60 +66,74 @@ type EnvResolver interface {
 	// if the orgName parameter is empty the platform helm env home is returned
 	ResolveHelmEnv(ctx context.Context, organizationID uint) (HelmEnv, error)
 
+	// ResolvePlatformEnv resolves the helm environment dedicated for the platform user
 	ResolvePlatformEnv(ctx context.Context) (HelmEnv, error)
 }
 
-type helm2EnvResolver struct {
-	// helmHomes the configurable directory location where helm homes are to be set up
-	helmHomes  string
-	orgService OrgService
-	logger     Logger
-}
-
-func NewHelm2EnvResolver(helmHome string, orgService OrgService, logger Logger) EnvResolver {
-	return helm2EnvResolver{
-		helmHomes:  helmHome,
-		orgService: orgService,
-		logger:     logger,
-	}
+// envResolver generic env resolver to be embedded into EnvResolver implementations
+type envResolver struct {
+	helmHomesDir string
+	orgService   OrgService
+	logger       Logger
 }
 
-func (h2r helm2EnvResolver) ResolveHelmEnv(ctx context.Context, organizationID uint) (HelmEnv, error) {
-	h2r.logger.Debug("resolving organization helm env home")
-	orgName, err := h2r.orgService.GetOrgNameByOrgID(ctx, organizationID)
+func (er envResolver) ResolveHelmEnv(ctx context.Context, organizationID uint) (HelmEnv, error) {
+	er.logger.Debug("resolving organization helm env home")
+	orgName, err := er.orgService.GetOrgNameByOrgID(ctx, organizationID)
 	if err != nil {
 		return HelmEnv{}, errors.WrapIfWithDetails(err, "failed to get organization name for ID",
 			"organizationID", organizationID)
 	}
 
 	return HelmEnv{
-		home:     path.Join(h2r.helmHomes, orgName, helmPostFix),
+		home:     path.Join(er.helmHomesDir, orgName, helmPostFix),
 		platform: false,
 	}, nil
 }
 
-func (h2r helm2EnvResolver) ResolvePlatformEnv(ctx context.Context) (HelmEnv, error) {
+func (er envResolver) ResolvePlatformEnv(ctx context.Context) (HelmEnv, error) {
 	return HelmEnv{
-		home:     path.Join(h2r.helmHomes, PlatformHelmHome, helmPostFix),
+		home:     path.Join(er.helmHomesDir, PlatformHelmHome, helmPostFix),
```

Comment:
What happens if this guy registers to the dev preview?

https://github.com/pipeline

Reply:
I will introduce a new configuration option for the platform home

Reply:
I chose a different approach: https://github.com/banzaicloud/pipeline/pull/2861/commits/6d66faf27d2067f035a9cb8ae4013e0e992bce2f